mesonbuild/compilers/c.py

```diff
@@ -219,9 +219,22 @@ def _get_search_dirs(self, env):
     def _split_fetch_real_dirs(pathstr, sep=':'):
         paths = []
         for p in pathstr.split(sep):
-            p = Path(p)
-            if p.exists() and p.resolve().as_posix() not in paths:
-                paths.append(p.resolve().as_posix())
+            # GCC returns paths like this:
+            # /usr/lib/gcc/x86_64-linux-gnu/8/../../../../x86_64-linux-gnu/lib
+            # It would make sense to normalize them to get rid of the .. parts
+            # Sadly when you are on a merged /usr fs it also kills these:
```

Comment:
I don't think this is quite accurate, no one on Fedora or Arch (which both have merged /usr) seem to have this problem. I think this problem is caused by compiling gcc with a split /usr and then using that gcc with a merged /usr

Reply:
Well we have to support this anyway since it's in the wild...